docs/explanations/performance.md

```diff
@@ -0,0 +1,124 @@
+# Napari Performance
+
+Performance is a core feature of napari. Performance is a concern for every
+feature that napari implements. Without adequate performance users will migrate
+to other visualization tools, even if napari has impressive functionality.
+
+There are two main types of performance:
+
+**Objective Performance**
+* How long operations take when timed with a stopwatch.
+* Most times will vary based on the data being viewed.
+
+**Subjective Performance**
+* The user’s experience as it relates to performance.
+* Is the user’s experience pleasant or frustrating? Does napari "seem fast"?
+
+Both types of performance are important. No amount of slickness can make up for
+an application that is fundamentally too slow. And even a relatively fast
+application can feel clunky or frustrating if not designed well.
+
+## Objective Performance
+
+How to keep napari objectively fast:
+
+**Focus On Real Use Cases**
+
+* Focus on cases that matter to lots of people.
+* It’s easy to waste time optimizing things no one cares about or no one will
+  notice.
+* If a dataset is unreasonable or out of scope or fringe, don’t waste time
+  trying to make it run fast.
+
+**Always Be Timing**
+* Build timers into the software that always run.
+* If not always visible, power users and developers should be able to toggle them on.
+* This gives people an ambient awareness of how long things take.
+* Allows users to report concrete performance numbers:
+  *  *it seemed slow* &#8594; *it ran at 10Hz*.
+  *  *it took a long time* &#8594; *it took 2 minutes and 30 seconds*.
+* Teaches users how different hardware impacts performance.
+
+**Performance System Tests**
+* Create automatic tests that time specific operations in specific known datasets.
+* Time many different operations on a nice selection of different datasets.
+
+**Performance Unit Tests**
+* Time one small operation to monitor for regressions.
+* Napari has some of these today as "benchmarks".
+* Interesting to see how different hardware performs as time goes on.
+
+**Run All Tests Every Merge**
+* Save results to a database or somewhere.
+* Catch a regression right when it happens and not weeks or
+  months later.
+* See how new features run on large datasets no one tested.
```

Comment:
These are great! Setting up continuous ASV testing would be wonderful.

Reply:
I added mention of ASV. In general though the document is pretty vague. Talking with @sofroniewn we kind of felt the order here is:
1. Write this grand performance vision document.
1. Put it aside and dive into obvious big problems like #845 and #204 
1. Come up for air and now survey the performance landscape and prioritize what to do, including automated tests, timing, benchmarking, etc.
1. At that point maybe this doc can point to lots of things we are actually doing to support these goals.

The point being #845 is so glaring it's worth tackling first.